src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -436,10 +440,14 @@ private function validateClassNode(object $object, ?string $cacheKey, ClassMetad
                     $group = $metadata->getGroupSequence();
                     $defaultOverridden = true;
                 } elseif ($metadata->isGroupSequenceProvider()) {
-                    // The group sequence is dynamically obtained from the validated
-                    // object
-                    /* @var \Symfony\Component\Validator\GroupSequenceProviderInterface $object */
-                    $group = $object->getGroupSequence();
+                    if (null !== $providerClass = $metadata->getGroupSequenceProviderClass()) {
+                        $group = $this->groupSequenceProviderFactory->getInstance($providerClass)->getGroupSequence();
```

Comment:
what about passing `$object` to `getGroupSequence`, for even more dynamic group computation.

this is what API platform currently does already: https://api-platform.com/docs/core/validation/#dynamic-validation-groups

Reply:
Good idea! I will update accordingly, thanks!